### ISSUE TITLE
[BUG] Unable to configure debug/trace logging for security plugins

### ISSUE THREAD

**Issue by patcable** [opensearch-project/security#1689]
**Describe the bug**
In the [Troubleshooting OpenID Connect](https://opensearch.org/docs/latest/troubleshoot/openid-connect) document it mentions that you can turn on trace logging for OpenID logging. I tried to use this with LDAP - replacing `com.amazon.dllc.auth.http.jwt` with `com.amazon.dlic.auth.ldap.backend`. Regardless of the value for `logger.plugins.security.name` when I try to start OS i get the following on stdout:
```
Mar 18 20:56:13 i-02720ef57ad2d3ad4 opensearch[93705]: Exception in thread "main" org.apache.logging.log4j.core.config.ConfigurationException: No name attribute provided for Logger plugins
Mar 18 20:56:13 i-02720ef57ad2d3ad4 opensearch[93705]:         at org.apache.logging.log4j.core.config.properties.PropertiesConfigurationBuilder.createLogger(PropertiesConfigurationBuilder.java:256)
Mar 18 20:56:13 i-02720ef57ad2d3ad4 opensearch[93705]:         at org.apache.logging.log4j.core.config.properties.PropertiesConfigurationBuilder.build(PropertiesConfigurationBuilder.java:178)
Mar 18 20:56:13 i-02720ef57ad2d3ad4 opensearch[93705]:         at org.opensearch.common.logging.LogConfigurator$2.getConfiguration(LogConfigurator.java:228)
Mar 18 20:56:13 i-02720ef57ad2d3ad4 opensearch[93705]:         at org.opensearch.common.logging.LogConfigurator$2.getConfiguration(LogConfigurator.java:197)
Mar 18 20:56:13 i-02720ef57ad2d3ad4 opensearch[93705]:         at org.apache.logging.log4j.core.config.ConfigurationFactory.getConfiguration(ConfigurationFactory.java:302)
Mar 18 20:56:13 i-02720ef57ad2d3ad4 opensearch[93705]:         at org.opensearch.common.logging.LogConfigurator$3.visitFile(LogConfigurator.java:236)
Mar 18 20:56:13 i-02720ef57ad2d3ad4 opensearch[93705]:         at org.opensearch.common.logging.LogConfigurator$3.visitFile(LogConfigurator.java:232)
Mar 18 20:56:13 i-02720ef57ad2d3ad4 opensearch[93705]:         at java.base/java.nio.file.Files.walkFileTree(Files.java:2725)
Mar 18 20:56:13 i-02720ef57ad2d3ad4 opensearch[93705]:         at org.opensearch.common.logging.LogConfigurator.configure(LogConfigurator.java:232)
Mar 18 20:56:13 i-02720ef57ad2d3ad4 opensearch[93705]:         at org.opensearch.common.logging.LogConfigurator.configure(LogConfigurator.java:142)
Mar 18 20:56:13 i-02720ef57ad2d3ad4 opensearch[93705]:         at org.opensearch.bootstrap.Bootstrap.init(Bootstrap.java:373)
Mar 18 20:56:13 i-02720ef57ad2d3ad4 opensearch[93705]:         at org.opensearch.bootstrap.OpenSearch.init(OpenSearch.java:178)
Mar 18 20:56:13 i-02720ef57ad2d3ad4 opensearch[93705]:         at org.opensearch.bootstrap.OpenSearch.execute(OpenSearch.java:169)
Mar 18 20:56:13 i-02720ef57ad2d3ad4 opensearch[93705]:         at org.opensearch.cli.EnvironmentAwareCommand.execute(EnvironmentAwareCommand.java:100)
Mar 18 20:56:13 i-02720ef57ad2d3ad4 opensearch[93705]:         at org.opensearch.cli.Command.mainWithoutErrorHandling(Command.java:138)
Mar 18 20:56:13 i-02720ef57ad2d3ad4 opensearch[93705]:         at org.opensearch.cli.Command.main(Command.java:101)
Mar 18 20:56:13 i-02720ef57ad2d3ad4 opensearch[93705]:         at org.opensearch.bootstrap.OpenSearch.main(OpenSearch.java:135)
Mar 18 20:56:13 i-02720ef57ad2d3ad4 opensearch[93705]:         at org.opensearch.bootstrap.OpenSearch.main(OpenSearch.java:101)
```

**To Reproduce**
Add the following to the log4j properties config (in my case, `/etc/opensearch/log4j2.properties`):
```
logger.plugins.security.name = com.amazon.dlic.auth.http.jwt
logger.plugins.security.level = trace
```
then try to start opensearch. You'll get the message above.

**Expected behavior**
I get debug logging for security modules, ideally the LDAP one since that's what I want to debug.

**Plugins**
```
[2022-03-18T20:56:39,763][INFO ][o.o.p.PluginsService     ] [i-02720ef57ad2d3ad4] loaded plugin [discovery-ec2]
[2022-03-18T20:56:39,763][INFO ][o.o.p.PluginsService     ] [i-02720ef57ad2d3ad4] loaded plugin [opensearch-alerting]
[2022-03-18T20:56:39,763][INFO ][o.o.p.PluginsService     ] [i-02720ef57ad2d3ad4] loaded plugin [opensearch-anomaly-detection]
[2022-03-18T20:56:39,763][INFO ][o.o.p.PluginsService     ] [i-02720ef57ad2d3ad4] loaded plugin [opensearch-asynchronous-search]
[2022-03-18T20:56:39,763][INFO ][o.o.p.PluginsService     ] [i-02720ef57ad2d3ad4] loaded plugin [opensearch-cross-cluster-replication]
[2022-03-18T20:56:39,763][INFO ][o.o.p.PluginsService     ] [i-02720ef57ad2d3ad4] loaded plugin [opensearch-index-management]
[2022-03-18T20:56:39,763][INFO ][o.o.p.PluginsService     ] [i-02720ef57ad2d3ad4] loaded plugin [opensearch-job-scheduler]
[2022-03-18T20:56:39,764][INFO ][o.o.p.PluginsService     ] [i-02720ef57ad2d3ad4] loaded plugin [opensearch-knn]
[2022-03-18T20:56:39,764][INFO ][o.o.p.PluginsService     ] [i-02720ef57ad2d3ad4] loaded plugin [opensearch-ml]
[2022-03-18T20:56:39,764][INFO ][o.o.p.PluginsService     ] [i-02720ef57ad2d3ad4] loaded plugin [opensearch-observability]
[2022-03-18T20:56:39,764][INFO ][o.o.p.PluginsService     ] [i-02720ef57ad2d3ad4] loaded plugin [opensearch-performance-analyzer]
[2022-03-18T20:56:39,764][INFO ][o.o.p.PluginsService     ] [i-02720ef57ad2d3ad4] loaded plugin [opensearch-reports-scheduler]
[2022-03-18T20:56:39,764][INFO ][o.o.p.PluginsService     ] [i-02720ef57ad2d3ad4] loaded plugin [opensearch-security]
[2022-03-18T20:56:39,764][INFO ][o.o.p.PluginsService     ] [i-02720ef57ad2d3ad4] loaded plugin [opensearch-sql]
```

**Host/Environment:**
 - OS: Ubuntu 20.04
 - Version: OpenSearch 1.3.0

**Comment by mmiller1** [comment]
I would also appreciate insight on how to enable debug logging for the security plugin, specifically LDAP as OP mentioned. I'm running the same OS / ES version as well.

**Comment by mmiller1** [comment]
I believe logging for the security plugin is just broken in 1.3.0, the following warning is observed when starting elasticsearch:
SLF4J: Failed to load class "org.slf4j.impl.StaticLoggerBinder"
SLF4J: Defaulting to no-operation (NOP) logger implementation

I installed version 1.2.4 and have been able to see the security plugin logs to debug my implementation.

**Comment by peternied** [comment]
Thanks for filing @patcable I am going to look into this, we've got a theory that it could be related to https://github.com/opensearch-project/security/pull/1563 that was added in 1.3.0

**Comment by peternied** [comment]
For getting a fix out it would likely be part of our [2.0.0 release](https://github.com/opensearch-project/opensearch-build/issues/1624).  I think @mmiller1 's suggestion to stay with 1.2.4 is solid.

**Comment by patcable** [comment]
Sounds good. Thanks for the context!

**Comment by mmiller1** [comment]
For what it's worth, with logging working in 1.2.4, we were able to fix our security plugin configuration and did upgrade back to 1.3.0 afterwards and things remained in a functional state (despite not have logs). This probably isn't ideal for production use, but because we're simply POC'ing opensearch, this is the route we took.

**Comment by peternied** [comment]
Adding some findings from my investigation, first of all here is how I am reproducing:

```
wget https://artifacts.opensearch.org/releases/bundle/opensearch/1.0.0/opensearch-1.0.0-linux-x64.tar.gz
tar -xvf opensearch-1.0.0-linux-x64.tar.gz
cd opensearch-1.0.0 
echo "logger.plugins.security.name = com.amazon.dlic.auth.http.jwt" >> config/log4j2.properties
echo "logger.plugins.security.level = trace" >> config/log4j2.properties
./opensearch-tar-install.sh
```

The behavior that I am seeing is for `1.0.0`, `1.2.4` and `1.3.0` the behavior is the same, as what is in the description of this issue.

**Comment by peternied** [comment]
Circling back! Good news, this looks like a documentation issue, not a product bug, could you try the following configuration in your `config/log4j2.properties` file and see if the logging appears as you'd expect?

```
logger.securityjwt.name = com.amazon.dlic.auth.http.jwt
logger.securityjwt.security.level = trace
```

Per the [yml format](https://logging.apache.org/log4j/2.x/manual/configuration.html#Configuration_with_Properties) in log4j properties, a logger is declared with `logger.LOGGER_NAME.name`, the value of LOGGER_NAME can be updated as you desire.  

When the example included the `.` operator it was treated as another object and 'security' was referenced as a key which didn't resolve to anything, that error message wasn't great because it should have been `No name attribute provided for Logger named 'plugins'`.

I'll create a pull request to update the documentation, it looks like several others have been impacted by this issue in the past

**Comment by patcable** [comment]
Hmm. Will this work for the LDAP backend @peternied? That's where I actually need debug output from. I'll give the JWT example a go to confirm it works though.

**Comment by patcable** [comment]
Just tested -- OpenSearch loads with that syntax, so that's encouraging 🎉 

But yeah, back to my original issue -- if I wanted to get LDAP logs out of 1.3.0 so I could understand why my authz/n config isnt working is there a way for that? or is it per package and i'm best off going to 1.2.4 then back to 1.3.0?

**Comment by peternied** [comment]
@patcable  Yes you can use `com.amazon.dlic.auth.ldap.backend` as the logger path and it let you adjust the tracing levels or send the logs to another appender

**Comment by patcable** [comment]
Hm. No dice for me, which makes me think there's a config issue. Happy to take this elsewhere, too. Is there a more global logger path I could use for the security module as a whole? Internal auth seems to work fine, fwiw.

**Comment by peternied** [comment]
@patcable What about setting the default logging level to tracing?  I wouldn't recommend doing that in prod or a disk space constrained environment

**Comment by patcable** [comment]
@peternied had a chance to give that a try just now (setting `rootLogger.level = trace`). There's no info here from the security module it would seem. Security module did helpfully ask OS to write a record of the failure at least?

```
[2022-04-01T17:42:05,803][TRACE][o.o.h.AbstractHttpServerTransport] [i-029996662e6220c9f] Http channel accepted: Netty4HttpChannel{localAddress=/10.0.16.13:9200, remoteAddress=/10.0.16.13:55072}
[2022-04-01T17:42:05,826][DEBUG][i.n.h.s.SslHandler       ] [i-029996662e6220c9f] [id: 0x4279ae99, L:/10.0.16.13:9200 - R:/10.0.16.13:55072] HANDSHAKEN: protocol:TLSv1.3 cipher suite:TLS_AES_256_GCM_SHA384
[2022-04-01T17:42:05,826][TRACE][o.o.h.HttpTracer         ] [i-029996662e6220c9f] [2][null][GET][/] received request from [Netty4HttpChannel{localAddress=/10.0.16.13:9200, remoteAddress=/10.0.16.13:55072}]
[2022-04-01T17:42:05,827][TRACE][o.o.i.b.in_flight_requests] [i-029996662e6220c9f] [in_flight_requests] Adding [0b][<http_request>] to used bytes [new used: [0b], limit: 4048289792 [3.7gb], estimate: 0 [0b]]
[2022-04-01T17:42:05,834][TRACE][o.o.i.b.in_flight_requests] [i-029996662e6220c9f] [in_flight_requests] Adjusted breaker by [0] bytes, now [0]
[2022-04-01T17:42:05,835][TRACE][o.o.h.HttpTracer         ] [i-029996662e6220c9f] [2][null][UNAUTHORIZED][text/plain; charset=UTF-8][12] sent response to [Netty4HttpChannel{localAddress=/10.0.16.13:9200, remoteAddress=/10.0.16.13:55072}] success [true]
[2022-04-01T17:42:05,836][TRACE][o.o.t.TaskManager        ] [i-029996662e6220c9f] register 150 [transport] [indices:data/write/index] [index {[security-auditlog-2022.04.01][_doc][null], source[{"audit_cluster_name":"lab","audit_node_name":"i-029996662e6220c9f","audit_rest_request_method":"GET","audit_category":"FAILED_LOGIN","audit_request_origin":"REST","audit_node_id":"Ehv5GTsLSRW4Krrb8b2NeA","audit_request_layer":"REST","audit_rest_request_path":"/","@timestamp":"2022-04-01T17:42:05.835+00:00","audit_request_effective_user_is_admin":false,"audit_format_version":4,"audit_request_remote_address":"10.0.16.13","audit_node_host_address":"10.0.16.13","audit_rest_request_headers":{"User-Agent":["curl/7.68.0"],"content-length":["0"],"Host":["i-029996662e6220c9f.example.com:9200"],"Accept":["*/*"]},"audit_request_effective_user":"cable","audit_node_host_name":"10.0.16.13"}]}]
[2022-04-01T17:42:05,837][TRACE][o.o.t.TaskManager        ] [i-029996662e6220c9f] register 151 [transport] [indices:data/write/bulk[s]] [requests[1], index[security-auditlog-2022.04.01][0], refresh[IMMEDIATE]]
```

**Comment by peternied** [comment]
@bublikhpt thanks for the ping, we are no longer packaging `log4j-slf4j-impl-X.XX.X.jar` with the security plugin.  I did a quick pilot, by adding back the library we are getting log messages again.

I'm doing background for the right place to make sure this library is included

**Comment by peternied** [comment]
Reopening - I'd like to make sure that we are seeing the fix before assuming its working as intended.

**Comment by peternied** [comment]
I was able to use one of our most recent builds of OpenSearch 2.0.0 and I see that the fix is operational, tested distribution: https://ci.opensearch.org/ci/dbc/distribution-build-opensearch/2.0.0-alpha1/2002/linux/x64/dist/opensearch/opensearch-2.0.0-alpha1-linux-x64.tar.gz

I am closing this issue out, we will be backporting this fix into 1.3.X builds which you can track with https://github.com/opensearch-project/security/issues/1753

**Comment by idiaz-at-globant** [comment]
Came here and spent two days figuring out why 1.3.x isn't logging the security plugin.

# rootLogger.level = info
# rootLogger.level = trace
# rootLogger.appenderRef.console.ref = console

# Debug security
logger.securityjwt.name = com.amazon.dlic.auth.http.jwt
logger.securityjwt.level = trace
logger.securityjwt.appenderRef.console.ref = console
I'm not a java expert, assuming you can uncomment the rootLogger

This how this works for me, the documentation doesn't work.

Using helm chats v1.18.0. Hope this help for anyone who came here.
